### PR TITLE
Update to SDK 0.46.14, ibc-go 6.2.0, latest gravity

### PR DESCRIPTION
**Pull request by EricBolten**
* Update to SDK 0.46.14
* Update to ibc-go 6.2.0
* Update to the latest commit on gravity-bridge main
* A large number of fixes for the new APIs of the SDK/ibc-go

Some things to note:
* Updating the way app options are done in `cmd/sommelier/cmd/root.go` could use a sanity check
* Updating the params for the ICA host in the v7 upgrade to include all message types -- is there a reason we would want to stick to the old list?

### CODE REVIEW

File: cmd/sommelier/cmd/root.go [hunk 1]
```diff
@@ -178,49 +175,27 @@ func txCommand() *cobra.Command {
...
-		panic(err)
-	}
+	// TODO(bolten): sanity check this update
+	// appears to do exactly the same things we were manually doing before, and also
+	// includes new baseapp options:
+	//
+	// baseapp.SetIAVLCacheSize(cast.ToInt(appOpts.Get(FlagIAVLCacheSize))),
+	// baseapp.SetIAVLDisableFastNode(cast.ToBool(appOpts.Get(FlagDisableIAVLFastNode))),
+	// baseapp.SetIAVLLazyLoading(cast.ToBool(appOpts.Get(FlagIAVLLazyLoading))),
+	baseappOptions := server.DefaultBaseappOptions(appOpts)
```
Comment (cbrit) [inline cmd/sommelier/cmd/root.go:190]: The only one I couldn't find (in this file) a mapping for was `SetQueryGasLimit`, which is in the new defaults, but other than that it seems correct to me.

**Review by cbrit — COMMENTED** [review]
One comment, will help investigate the tests

File: app/upgrades/v7/upgrades.go [hunk 2]
```diff
@@ -17,11 +19,18 @@ func CreateUpgradeHandler(
...
 	pubsubKeeper pubsubkeeper.Keeper,
 ) upgradetypes.UpgradeHandler {
 	return func(ctx sdk.Context, plan upgradetypes.Plan, fromVM module.VersionMap) (module.VersionMap, error) {
 		ctx.Logger().Info("v7 upgrade: entering handler")
 
+		// TODO(bolten): get a sanity check on this
+		// Now that we're on IBC V6, we can update the ICA host module to allow all message types rather than
+		// the list we specified in the v6 upgrade -- a default of HostEnabled: true and the string "*" for messages
+		icaParams := icahosttypes.DefaultParams()
+		icaHostKeeper.SetParams(ctx, icaParams)
```
Comment (zmanian) [inline app/upgrades/v7/upgrades.go:32]: Looks correct

https://github.com/cosmos/ibc-go/blob/v6.2.1/modules/apps/27-interchain-accounts/host/types/keys.go#L15

Comment (EricBolten) [reply app/upgrades/v7/upgrades.go:32]: Yeah, I think my concern is more whether or not we should allow all messages or stick with the original list we had. Any risk you think?

Comment (zmanian) [reply app/upgrades/v7/upgrades.go:32]: I think all messages are fine

**Review by zmanian — APPROVED** [review]

**Pull request merged**